src/main/java/gregtech/api/items/toolitem/IGTTool.java

```diff
@@ -304,10 +325,16 @@ default float getTotalAttackSpeed(ItemStack stack) {
 
     default int getTotalMaxDurability(ItemStack stack) {
         NBTTagCompound toolTag = getToolTag(stack);
+        IGTToolDefinition toolStats = getToolStats();
         if (toolTag.hasKey(MAX_DURABILITY_KEY, Constants.NBT.TAG_INT)) {
             return toolTag.getInteger(MAX_DURABILITY_KEY);
         }
-        int maxDurability = getMaterialDurability(stack) + getToolStats().getBaseDurability(stack);
+        int maxAOE = Math.max(getMaxAoEDefinition(stack).column, Math.max(getMaxAoEDefinition(stack).row, getMaxAoEDefinition(stack).layer));
```

Comment:
`getMaxAoEDefinition` is costly as it is reads data from NBT and you are calling it here 3x times. You should call it only once, save it to local variable and then get column/row/layer from it.